src/currencyConversion/BitmamaAPI.ts

```diff
@@ -0,0 +1,60 @@
+import { RESTDataSource } from 'apollo-datasource-rest'
+import BigNumber from 'bignumber.js'
+import { logger } from '../logger'
+import { BITMAMA_API } from '../config'
+
+interface BitmamaAPIResult {
+  buy: number
+  sell: number
+}
+
+// ttl in seconds!
+const MIN_TTL = 12 * 3600 // 12 hours
+export default class BitmamaAPI extends RESTDataSource {
+  private rateCache: Map<string, BigNumber>
+  constructor() {
+    super()
+    this.baseURL = BITMAMA_API
+    this.rateCache = new Map<string, BigNumber>()
+  }
+
+  async getExchangeRate({
+    fromCurrencyCode,
+    toCurrencyCode,
+  }: {
+    fromCurrencyCode: string
+    toCurrencyCode: string
+  }): Promise<BigNumber> {
+    try {
+      const result = await this.get<BitmamaAPIResult>(
+        '/rates/fiat',
+        {
+          from: fromCurrencyCode,
+          to: toCurrencyCode,
+        },
+        {
+          cacheOptions: {
+            ttl: MIN_TTL, // Only actually call the API every MIN_TTL minutes so it doesn't get spammed
+          },
+        },
+      )
+      this.rateCache.set(
+        `${fromCurrencyCode}-${toCurrencyCode}`,
+        new BigNumber(result.buy),
+      )
+      return new BigNumber(result.buy)
+    } catch (error) {
+      logger.error({
+        type: 'ERROR_FETCHING_BITMAMA_EXCHANGE_RATE',
+        error,
+      })
+      // As a fallback, try to return the last cached rate
+      if (this.rateCache.has(`${fromCurrencyCode}-${toCurrencyCode}`)) {
+        return this.rateCache.get(
+          `${fromCurrencyCode}-${toCurrencyCode}`,
+        ) as BigNumber
```

Comment:
```suggestion
      const cachedRate = this.rateCache.get(`${fromCurrencyCode}-${toCurrencyCode}`)
      if (cachedRate) {
        return cachedRate
```
code golf. to avoid the type cast